src/test_l1_dsVideoDevice.c

```diff
@@ -507,34 +506,36 @@ void test_l1_dsVideoDevice_positive_dsGetDFC(void)
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     // Define variables
     int result;
-    int index = 0;
     intptr_t handle = -1;
-    dsVideoZoom_t dfc_mode_1;
-    dsVideoZoom_t dfc_mode_2;
+    dsVideoZoom_t dfc_mode;
 
     // Step 01: Initialize video devices
     result = dsVideoDeviceInit();
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
+    UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
     // Step 02: Obtain video device handle
-    result = dsGetVideoDevice(index, &handle);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-    UT_ASSERT_EQUAL((handle >= 0), true);
-
-    // Step 03: Get the DFC mode using dsGetDFC() with the obtained handle
-    result = dsGetDFC(handle, &dfc_mode_1);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-
-    // Step 04: Get the DFC mode using dsGetDFC() again
-    result = dsGetDFC(handle, &dfc_mode_2);
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
-
-    // Step 05: Compare the results to make sure they match
-    UT_ASSERT_EQUAL(memcmp(&dfc_mode_1, &dfc_mode_2, sizeof(dsVideoZoom_t)),0);
+    for(int i = 0; i < gDSvideoDevice_NumVideoDevices; i++){
+        result = dsGetVideoDevice(i, &handle);
+        UT_ASSERT_EQUAL(result, dsERR_NONE);
+        UT_ASSERT_NOT_EQUAL(handle, (intptr_t)NULL);
+
+        // Step 03: Get the DFC mode using dsGetDFC() with the obtained handle
+        if(gSourceType == 1) {
+            result = dsGetDFC(handle, &dfc_mode);
+            UT_ASSERT_EQUAL(result, dsERR_NONE);
+
+            // Step 04: Compare the DFC mode with value from the profile
+            for (int j = 0; j < gDSVideoDeviceConfiguration[j].NoOfSupportedDFCs; ++j) {
+                UT_ASSERT_EQUAL(dfc_mode, gDSVideoDeviceConfiguration[i].SupportedDFCs[j]);
```

Comment:
let say here dsGetDFC(handle, &dfc_mode); is return dfc_mode = dsVIDEO_ZOOM_FULL
We need to check dsVIDEO_ZOOM_FULL is available in the supported list 
but logic here is fails because dfc_mode  is also compare the dsVIDEO_ZOOM_PLATFORM and test case also fails